packages/sitecore-jss-nextjs/src/middleware/personalize-middleware.ts

```diff
@@ -211,10 +211,15 @@ export class PersonalizeMiddleware {
     // Set browserId cookie on the response
     this.setBrowserId(response, browserId);
 
+    const resHeaders: { [key: string]: string } = {};
+
+    // Extract headers to avoid middleware issue https://github.com/vercel/next.js/issues/39765
+    response.headers.forEach((value, key) => (resHeaders[key] = value));
+
```

Comment:
Maybe move this logic to a protected helper function similar to native-fetcher for consistency.